addons/mail/models/mail_channel.py

```diff
@@ -168,6 +173,12 @@ def _compute_member_count(self):
         for channel in self:
             channel.member_count = member_count_by_channel_id.get(channel.id, 0)
 
+    @api.depends('channel_type')
+    def _compute_group_public_id(self):
+        group_user = self.env.ref('base.group_user')
+        for channel in self:
+            channel.group_public_id = group_user if channel.channel_type == 'channel' else None
```

Comment:
You should not erase a previously set value. I would do something less invasive and batch-enabled like
```
@api.depends('channel_type')
def _compute_group_public_id(self):
    channels = self.filtered(lambda channel: channel.channel.type == 'channel')
    channels.group_public_id = False
    toupdate = (self - channels).filtered(lambda channel: not channel.group_public_id)
    toupdate.group_public_id = group_user
```